GenericNode.ts

```diff
@@ -0,0 +1,73 @@
+export class GenericNode {
+
+    /**
+     * Member which signifies what type of node this is.
+     */
+    private nodeType : GenericNode; //null for sheet
+
+    /**
+     * Member which contains the list of children nodes nested within this node.
+     */
+    private children : GenericNode[]; //for atoms, this becomes null because atom is always a leaf   
```

Comment:
Since this is null if the node is an atom you implicitly don't need it on the generic node, you only need it on cut nodes.

Reply:
Wouldn't it be better to have this here so that the sheet of assertion of the AEG tree can implement it as well? Because we were assuming that the sheet would be a generic node

Reply:
Since atom nodes don't have children, the parent class shouldn't have children, this is classic OOP design. Since we're doing OOP for this, it would make more sense to have the sheet either be a cut node and write the AEG methods around the assumption that the root of the entire AEG isn't a cut OR implement a ConjunctNode that inherits from GenericNode as the parent of cut node that represents a sheet. So the tree would look like:

```
           / > ConjunctNode -> CutNode
GenericNode 
           \ > Atom Node
```